conans/test/unittests/client/build/cpp_std_flags_test.py

```diff
@@ -1,195 +1,261 @@
-
-        self.assertEqual(cppstd_flag("gcc", "4.7", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "4.7", "14"), None)
+def make_cppstd_flag(compiler, compiler_version, cppstd):
+    settings = Settings.loads(get_default_settings_yml())
+    settings.cppstd = cppstd
+    settings.compiler = compiler
+    settings.compiler.version = compiler_version
+    settings.compiler.cppstd = cppstd
+    return cppstd_flag(settings)
 
-        self.assertEqual(cppstd_flag("gcc", "4.8", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "4.8", "14"), '-std=c++1y')
-        self.assertEqual(cppstd_flag("gcc", "4.8", "17"), None)
 
-        self.assertEqual(cppstd_flag("gcc", "4.9", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "4.9", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "4.9", "17"), None)
+def make_cppstd_default(compiler, compiler_version):
+    settings = Settings.loads(get_default_settings_yml())
+    settings.compiler = compiler
+    settings.compiler.version = compiler_version
+    return cppstd_default(settings)
 
-        self.assertEqual(cppstd_flag("gcc", "5", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "5", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "5", "gnu14"), '-std=gnu++14')
-        self.assertEqual(cppstd_flag("gcc", "5", "17"), None)
 
-        self.assertEqual(cppstd_flag("gcc", "5.1", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "5.1", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "5.1", "17"), '-std=c++1z')
-
-        self.assertEqual(cppstd_flag("gcc", "7", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "7", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "7", "17"), '-std=c++17')
+class CompilerFlagsTest(unittest.TestCase):
 
-        self.assertEqual(cppstd_flag("gcc", "8", "11"), '-std=c++11')
-        self.assertEqual(cppstd_flag("gcc", "8", "14"), '-std=c++14')
-        self.assertEqual(cppstd_flag("gcc", "8", "17"), '-std=c++17')
-        self.assertEqual(cppstd_flag("gcc", "8", "20"), '-std=c++2a')
+    def test_gcc_cppstd_flags(self):
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.2", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.2", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.2", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.2", "14")
+
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.3", "98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.3", "gnu98")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.3", "11")
+        with self.assertRaises(ConanException):
+            make_cppstd_flag("gcc", "4.3", "14")
```

Comment:
Failing because of the helper function, not because of the underlying `cppstd_flag` being tested here

```suggestion
```

Reply:
removed